src/sensors/IMD.py

```diff
@@ -0,0 +1,74 @@
+from pin.pinout import Pinout
+from services.PWM import PWM
+from services.digital_in import DigitalInService
+from shared_memory import SharedMemory
+from test_lib.input.aggregate import Multiple
+from test_lib.input.input import Input
+from test_lib.input.timed import After
+from enum import Enum
+
+class Case(Enum):
+    ShortCircuit    = 0
+    Normal          = 10
+    UnderVoltage    = 20
+    EquipmentFault  = 40
+    GroundFault     = 50
+
+class IMDInput(Input):
+    def __init__(self, frequency : int, duty_cycle : int):
+        self.frequency = frequency
+        self.duty_cycle = duty_cycle
+
+    async def apply(self):
+        self.m_hs.set_frequency(self.frequency)
+        self.m_hs.set_duty_cycle(self.duty_cycle)
+
+class IMD:
+    def __init__(self, shm:SharedMemory, ok_hs:Pinout, m_hs:Pinout):
+        self.ok_hs = DigitalInService(shm, ok_hs)
+        self.m_hs = PWM(shm, m_hs)
```

Comment:
```suggestion
        self.m_hs = InputCapture(shm, m_hs)
```
The logic mirrors the C++ structure, so you generate signals for an InputCapture, not for a PWM